lib/Backend/Opnd.h

```diff
@@ -1467,6 +1472,105 @@ class AddrOpnd sealed : public Opnd
 #endif
 };
 
+
+typedef RegOpnd ListOpndType;
+
+template<typename... T>
+struct ListOpndInit
+{
+    template<typename K1, typename... K>
+    void insert(int index, K1 arg, K... rest)
+    {
+        values[index] = arg;
+        insert(index + 1, rest...);
+    }
+    template<typename K>
+    void insert(int index, K last)
+    {
+        values[index] = last;
```

Comment:
>values[index] = last; [](start = 8, length = 21)

Is overwriting allowed?

Reply:
No, this structure should be a type safe variable arguments buffer that can be optimized away at compile time.
@penguinwizzard originally wrote this struct.
I was thinking of making this generic for possibly other purposes.
Also, you're making me realize that the insert method should probably be private

Reply:
You could probably even go one step further and make ListOpndInit a private class inside ListOpnd.